### PR TITLE
global: migrate CI to gh-actions

### LINKED ISSUE #92: global: migration to github-actions from travis

### PR DESCRIPTION
**Pull request by ParthS007**
closes #92

### CODE REVIEW

**Comment by diegodelemos** [comment]
@ParthS007 [Webpack support was introduced in Invenio-Assets `1.1`](https://github.com/inveniosoftware/invenio-assets/commit/453fc05efd03eba26e564603158e70db6f1a549b), could you try upgrading its minimum version to `1.1`? It that doesn't work try `1.2`. I think that with that we can make the GH actions build to pass.

File: run-tests.sh [hunk 1]
```diff
@@ -6,11 +6,17 @@
...
-python setup.py test && \
-sphinx-build -qnNW -b doctest docs docs/_build/doctest
+# Quit on errors
+set -o errexit
+
+# Quit on unbound symbols
+set -o nounset
+
+python setup.py compile_catalog
+pybabel compile -d tests/translations/
```
Comment (diegodelemos) [inline run-tests.sh:16]: Is there a way of invoking `pybabel` and `compile_catalog` the `python -m` way? We should try to always do it for tests reproducibility.

File: invenio_i18n/ext.py [hunk 2]
```diff
@@ -137,8 +137,8 @@ def iter_languages(self):
 
         yield (default_lang, default_title)
 
-        for l, title in current_app.config.get('I18N_LANGUAGES', []):
-            yield l, title
+        for language, title in current_app.config.get('I18N_LANGUAGES', []):
```
Comment (ppanero) [inline invenio_i18n/ext.py:140]: OCD: sometimes `language`, sometimes `lan`, sometimes `lang`. Would be nice, as long as scopes allow to have always the same naming

File: run-tests.sh (hunk 1 above)
Comment (ParthS007) [reply run-tests.sh:16]: `compile_catalog` is executing by `python -m` not `pybabel`.

File: invenio_i18n/ext.py (hunk 2 above)
Comment (ParthS007) [reply invenio_i18n/ext.py]: Addressed 👍

File: run-tests.sh (hunk 1 above)
Comment (ppanero) [reply run-tests.sh]: Don't they do the same? both seem to endup [here](https://github.com/python-babel/babel/blob/0a3402594d66eb4c8f76c42563a91a8ec88c3b04/babel/messages/frontend.py#L129)

Comment (diegodelemos) [reply run-tests.sh]: When I said invoking `pybabel` and `compile_catalog` the `python -m` way I meant:
```diff
- python setup.py compile_catalog
+ python -m setup compile_catalog
- pybabel compile -d tests/translations/
+ python -m babel.messages.frontend compile -d tests/translations/
```
So all commands are called inside the virtualenv.

Comment (ppanero) [reply run-tests.sh]: @diegodelemos I understood, then I wondered if it was needed. Cuz they both seem to end up calling the same function 🤔. But anyway let's leave it as it was (both calls), with `python -m...` if possible.

Comment (diegodelemos) [reply run-tests.sh]: @ppanero my message was addressing Parth's comment https://github.com/inveniosoftware/invenio-i18n/pull/93#discussion_r537406386, however, you posted while I was writing! I agree, if the commands are doing exactly the same, please remove one of them.

Comment (ppanero) [reply run-tests.sh]: Concurrency issues 😅 I'd say:

- Leave it as is now.
- Open an issue to investigate the possibility of removing one of the two commands. In order not to overlad @lnielsen right now.

@ParthS007  WDYT? 
@diegodelemos If this is also fine with you, feel free to merge.

**Review by diegodelemos — APPROVED** [review]

File: run-tests.sh (hunk 1 above)
Comment (ParthS007) [reply run-tests.sh]: Okay, I have opened another [issue](https://github.com/inveniosoftware/invenio-i18n/issues/95) and we can go ahead with merging this one.

**Pull request merged**